### ISSUE TITLE
Potential bug on listener logprobs

### ISSUE THREAD

**Issue by JACKHAHA363** [Joshua-Ren/Neural_Iterated_Learning#2]
https://github.com/Joshua-Ren/Neural_Iterated_Learning/blob/master/train.py#L93

In this line, the listener logprobs is computed on the batch axis, which does not seem correct?

The correct one would be:
F.log_softmax(pred_vector, dim=1).max(dim=1)[0]

**Comment by Joshua-Ren** [comment]
Hi,

Thanks very much for your email. And thanks for pointing out this bug in
the code. Actually, I also noticed this bug recently. But strangely, I
found this loss works so well. I have tried some different versions,
including the correct implementation (i.e.,
Softmax(pred_vector)[true_id].log().mean()), the cross-entropy loss between
logits and predictions, etc. But this loss performs best. Anyway, it is
also safe to use cross-entropy loss instead, which seems more reasonable.

Best,

Joshua

Yuchen Lu <notifications@github.com> 于2021年2月9日周二 上午5:11写道：
